go/cmd/vtctldclient/command/root.go

```diff
@@ -78,6 +79,26 @@ var (
 		// propagated).
 		SilenceErrors: true,
 		Version:       servenv.AppVersion.String(),
+		// If we've reached this function, it means that:
+		//
+		// (1) The user specified some positional arguments, which, for the way
+		// we've structured things can only be a subcommand name, **and**
+		//
+		// (2) Cobra was unable to find a subcommand with that name for which to
+		// call a Run or RunE function.
+		//
+		// From this we conclude that the user was trying to either run a
+		// command that doesn't exist (e.g. "vtctldclient delete-my-data") or
+		// has misspelled a legitimate command (e.g. "vtctldclient StapReplication").
+		// If we think this has happened, return an error, which will get
+		// displayed to the user in main.go along with the usage.
```

Comment:
Not something that I'm suggesting we bring in here but we could at some point use the "did you mean X" functionality that cobra provides via the `command.SuggestionsMinimumDistance = X` option. Maybe you explored it and there were downsides?

Reply:
i never explored it but i definitely want to at some point!